src/pages/Login.tsx

```diff
@@ -1,14 +1,45 @@
-import React from "react";
-import { withRouter } from "react-router-dom";
+import React, { useState, useEffect } from "react";
+import { RouteComponentProps, withRouter } from "react-router-dom";
+import md5 from "md5";
+import { useCookies } from "react-cookie";
 
 import { Button, Input, Typography } from "antd";
 
+import { API_DOMAIN } from "../config";
 import Header from "../components/Header";
 import "../styles/Login.css";
 
-const Login: React.FC = (props) => {
-  const onConfirmClick = () => {
-    // TODO: Write code here to try logging in (if successful - dashboard, if error then display error)
+const Login: React.FC<RouteComponentProps> = (props) => {
+  const [ username, setUsername ] = useState("");
+  const [ password, setPassword ] = useState("");
+  const [ cookies, setCookie ] = useCookies(['accessToken']);
+  const [ error, setError ] = useState(false);
+
+  useEffect(() => {
+    if (cookies['accessToken']) {
+      props.history.push("/Dashboard");
+    }
+  }, []);
+
+  const onConfirmClick = async (): Promise<void> => {
+    const hashedPassword = md5(password);
+    const response = await fetch(`${API_DOMAIN}users/authenticate/${username}`, {
+      method: "POST",
+      body: JSON.stringify({ passwordHash : hashedPassword }),
+      headers: {
+        'Content-Type': 'application/json',
+        'Accept': 'application/json',
+      },
+    });
+
+    const res = await response.json();
+    if (res && res.accessToken) {
+      setCookie('accessToken', res.accessToken);
+      setError(false);
+      props.history.push("/Dashboard");
+    } else {
+      setError(!error);
```

Comment:
hmm race conditions here could lead to some weird side effects.. best to explicitly set it as setError(true)